### ISSUE TITLE
Clicking "hop #somechannel" when right-clicking a channel mentioned in chat parts and rejoins if I'm already in the channel

### ISSUE THREAD

**Issue by ekolis** [kvirc/KVIrc#2058]
### Expected behavior

I'm taken to the other channel's chat window (it would be nice if just clicking the channel name did that)
### Actual behavior

I'm taken to the other channel's chat window, but the client parts and rejoins the channel
### Steps to reproduce the issue
1. Mention a channel in chat
2. Right click on it
3. Select "join #somechannel" from the popup menu
### System information

KVIrc 4.9.2 'Aria'

Runtime Info:
System name: Windows Release : 6.2.9200
System version: Windows 8 Professional (x64) (Build 9200)
Architecture: Intel64 Family 6 Model 37 Stepping 5
Qt version: 5.5.1
Qt theme: windowsvista

Build Info:
Build date: 2016-07-06 02:13:19 UTC
Sources date: 20160706Revision number: git-6779-g96e790b
System name: Windows-6.3.9600
CPU name: AMD64
Build command: C:/Program Files (x86)/CMake/bin/cmake.exe
Build flags: 
   MANDIR=share/man
   CMAKE_INSTALL_PREFIX=C:/projects/kvirc/build/release
   Threads=Win32
Compiler name: C:/Program Files (x86)/Microsoft Visual Studio 12.0/VC/bin/amd64/cl.exe
Compiler flags: /DWIN32 /D_WINDOWS /W3 /GR /EHsc
Qt version: 5.5.1
Features: IRC, IPv6, Crypt, SSL, IPC, Transparency, Webkit, Perl, Python, Enchant, Qt5, KVS

I'm running Windows 10, not Windows 8. Also my CPU is an Intel Core i3, not an AMD64...

**Comment by AndrioCelos** [comment]
Your KVIrc parts a channel you haven't joined yet?
I'm not seeing any problem here...

**Comment by ekolis** [comment]
No, I mean I'm in the channel already (in another subwindow), and KVIrc parts it and rejoins it.

**Comment by Ionic** [comment]
The bugs title doesn't match its description.

The title contains `hop <channel>`, where "hop" stands for `part and rejoin <channel>`, so what you see would exactly be expected behavior.

Your description talks about `join <channel>`, which, if KVIrc indeed does `part and rejoin <channel>`, isn't doing what it's supposed to be.

Please try to clear this inconsistency up.

**Comment by wodim** [comment]
The hop command does that, it parts and joins the channel. It's not meant to switch windows.

**Comment by ekolis** [comment]
Oh, I didn't realize that's what hop meant. Odd that there's no way to just
switch windows...

**Comment by wodim** [comment]
Yeah, click the window name on the window list...

**Comment by wodim** [comment]
You can though modify that popup menu through scripting and create a new entry that does that thing you want to do

**Comment by Ionic** [comment]
Double-clicking the channel name should actually make that window active, or am I mistaken?

**Comment by un1versal** [comment]
> Double-clicking the channel name should actually make that window active, or am I mistaken?

yes you are correct, but you can right click that channel name and select
- **hop** parts and rejoins any given channel (hop will also join a channel that youre not joined)
- **join** (shows up if the channel is not joined but doesn't show if you are joined)
- **part** (shows up if you are joined into that channel and will part that channel if you click that link even on some other channel on the same network.)

tl;dr joins and parts are **smart** and **show up conditionally** (these options know if youre joined or not something I added a while back) and act accordingly. 

**edit** for organization purpose Ill label this invalid

**Comment by Ionic** [comment]
I know. Just wanted to offer a solution to the OPs actual problem - jumping to the channel window. :)
